crates/wdk-build/src/metadata/mod.rs

```diff
@@ -173,3 +173,269 @@ pub(crate) fn iter_manifest_paths(metadata: Metadata) -> impl IntoIterator<Item
 
     cargo_manifest_paths
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Look for opportunities to reduce code duplication across tests. Duplicated code is not only a long term maintenance problem it makes it easy to miss things in the review as well.

Reply:
Thanks, I've refactored the tests and largely reduced the duplication. I think it should be easier to review now.